app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -290,6 +305,88 @@ private void selectPointer() {
         }
     }
 
+    /**
+     * Edit a message passed as parameter
+     *
+     * @param message Message to be edited
+     */
+    private void editMessage(final Message message, final int position) {
+        boolean isEditingAllowed = app.
+                getSettings().
+                getBoolean(Constants.IS_EDITING_ALLOWED, Constants.DEFAULT_EDITING_ALLOWED);
+        int maxMessageContentEditLimit = app.
+                getSettings().
+                getInt(Constants.MAXIMUM_CONTENT_EDIT_LIMIT, Constants.DEFAULT_MAXIMUM_CONTENT_EDIT_LIMIT);
+        int timeSinceMessageSend = (int) ((System.currentTimeMillis() - message.getTimestamp().getTime()) / 1000);
+
+        if (!isEditingAllowed) {
+            Toast.makeText(getContext(), R.string.editing_message_disabled, Toast.LENGTH_SHORT).show();
+        } else if (timeSinceMessageSend > maxMessageContentEditLimit) {
+            Toast.makeText(getContext(), R.string.maximum_time_limit_error, Toast.LENGTH_SHORT).show();
+        } else {
+            final View dialogView = View.inflate(getContext(),
+                    R.layout.message_edit_dialog, null);
+            //Pop up a dialog box with previous message content
+            final AlertDialog dialog = new AlertDialog.Builder(getContext())
+                    .setTitle(R.string.edit_message)
+                    .setView(dialogView)
+                    .setPositiveButton(android.R.string.ok, null)
+                    .setNegativeButton(android.R.string.cancel, null)
+                    .create();
+            dialog.getWindow().setSoftInputMode(
+                    WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
+            dialog.show();
+
+            final EditText dialogMessageEditText = (EditText) dialogView.findViewById(R.id.message_content);
+            dialogMessageEditText.setText(message.getContent().trim());
```

Comment:
@kunall17 We had to trim the message response because message content has new line at the end and if we click on `Edit Message`(without adding `trim()`) then it appears some thing like this:

![rsz_11](https://cloud.githubusercontent.com/assets/15157620/22395629/a71d52ca-e569-11e6-97d9-2288d7caec31.png)

After adding `trim()` is appears as usual

![rsz_2](https://cloud.githubusercontent.com/assets/15157620/22395630/aaaf48f8-e569-11e6-93fb-5f31408f00b0.png)